mernarezk/assignment2/BinaryTree.java

```diff
@@ -0,0 +1,57 @@
+package Assignment2;
+
+/**
+ * Simple Binary Tree class that takes int values since BOTH problems are
+ * defined by taking in values and for testing simplicity methods params in
+ * assignments take direct int's rather than node references.
+ */
+public class BinaryTree {
+	TreeNode root;
+
+	/**
+	 * For testing and demonstration purposes the isnertion method happens int the
+	 * Binary Search tree because it's more imagineable how the tree structure will

```

Comment:
In the problem description it says that you cannot make an assumption that the tree is a Binary Search tree.
Can you think of another way of constructing a tree? You could either allow the user  to create the tree by hand (creating a node and setting its value, creating two more nodes and setting them as the right and left child of the parent, setting the first node as parent for the two children etc...) or to be able to construct a tree based on a in-order, pre-order or post-order description of the tree (using brackets "(" and ")" for example). The second approach appears very often during interviews, so it would be a good exercise to practise it now. :)

Reply:
Oh yes! great idea that haven't crossed my mind while I am implementing. I construct the three now using inOrder + preOrder arrays.